### PR TITLE
Upgrade to version 1.4.0

### PR DESCRIPTION
**Pull request by hcho3**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by hcho3** [comment]
@conda-forge-admin, please rerender

File: recipe/0001-conda-Unbundle-libxgboost.-dll-dylib-so.patch [hunk 1]
```diff
@@ -1,8 +1,8 @@
 diff --git a/python-package/xgboost/libpath.py b/python-package/xgboost/libpath.py
-index 8ca53fac..7122110f 100644
+index f7a7d9cd..70d73626 100644
 --- a/python-package/xgboost/libpath.py
 +++ b/python-package/xgboost/libpath.py
-@@ -28,7 +28,10 @@ def find_lib_path():
+@@ -29,7 +29,10 @@ def find_lib_path() -> List[str]:
```
Comment (trivialfis) [inline recipe/0001-conda-Unbundle-libxgboost.-dll-dylib-so.patch:5]: Is this patch still necessary?

Comment (hcho3) [reply recipe/0001-conda-Unbundle-libxgboost.-dll-dylib-so.patch:5]: Yes. The OS type can be set to `msys`:
https://github.com/conda-forge/xgboost-feedstock/blob/d04e880fac1889bd1cf3f96029d02e955fba1ccc/recipe/install-libxgboost.sh#L9-L12

File: recipe/meta.yaml [hunk 2]
```diff
@@ -128,7 +128,7 @@ outputs:
         - r-matrix
         - r-data.table
         - r-magrittr
-        - r-stringi
+        - r-jsonlite
```
Comment (jakirkham) [inline recipe/meta.yaml:131]: Is `r-stringi` still needed?

Comment (hcho3) [reply recipe/meta.yaml:131]: No. This is why it's being removed.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Philip and Jiaming! 😄

**Pull request merged**